arsclib-utils/src/main/kotlin/app/revanced/arsc/resource/ResourceFile.kt

```diff
@@ -0,0 +1,86 @@
+package app.revanced.arsc.resource
+
+import app.revanced.arsc.ApkResourceException
+import app.revanced.arsc.archive.Archive
+import app.revanced.arsc.resource.ResourceFile.Handle
+import com.reandroid.xml.XMLDocument
+import com.reandroid.xml.XMLException
+import java.io.*
+
+/**
+ * Instantiate a [ResourceFile] and lock the file which [handle] is associated with.
+ *
+ * @param handle The [Handle] associated with this file.
+ * @param archive The [Archive] that the file resides in.
+ */
+class ResourceFile private constructor(
```

Comment:
This file should be removed in favor of the new class `ArchiveResource.` You use dependency inversion to inject the archive this resource belongs to and write back to it. This can be implemented in `ArchiveResource` too. Is it necessary to invert the dependency to `Handle`? Also, `ArchiveResource` is abstracted in a way that does not need to read all bytes to a byte array. Before my changes, you were reading all bytes from a parsed XmlDocument file, then later on parsing it again in DomFIleEditor as an w3c XML file. w3c XML files should be removed, and XmlDocument from ARSCLib should be used instead. This way, we stop depending on w3c XML as well as don't have to do all the unnecessary writing and parsing